hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/EmbeddedAttributeMapping.java

```diff
@@ -407,4 +403,9 @@ public boolean containsTableReference(String tableExpression) {
 		}
 		return producer.containsTableReference( tableExpression );
 	}
+
+	@Override
+	public int compare(Object value1, Object value2) {
```

Comment:
Not sure if necessary, but it seems that maybe `EmbeddedIdentifierMappingImpl` should receive a similar implementation for `compare`.

Reply:
Nice catch - maybe it's not currently used, but for consistency I think it should. I've added the same implementation for embedded identifiers.